### PR TITLE
Fix Docker compose healthcheck

### PR DESCRIPTION
**Pull request by Cronick**
gcr.io/distroless/static-debian11 doesn't include curl by default, and because it's distroless. There is not really an obvious way (from what I can see) to get it installed.

Proposal is to change base image to debian:stable-slim and install curl into the container.
This will increase the image size by 4x, from what I could test on my own machine, and not as "secure" as a distroless.

So just opening the discussion now, if we should have a working healthcheck or not 😄

### CODE REVIEW

**Comment by Fabio1988** [comment]
What about this article? 
https://medium.com/@aminmir326/health-checks-for-distroless-containers-a2180c4c4fcf

Might be a workaround

**Comment by na-ji** [comment]
What I usually do on distroless images is just to copy the curl binary from a busybox image. It's easy and work well. Keeping the advantage of distroless with little downside.

**Comment by Cronick** [comment]
> What I usually do on distroless images is just to copy the curl binary from a busybox image. It's easy and works well. Keeping the advantage of distroless with little downside.

Will give that an attempt tomorrow then, if no-one beats me to do it 😄

**Comment by comstud** [comment]
Oops. Didn’t think about this. I only tested that the route worked in the service. Copying curl binary would be the better solution IMO

**Comment by lenisko** [comment]
Guilty! I'm not running Fletchling inside a docker 😶

**Comment by lenisko** [comment]
@Cronick you might want to use that solution (well with maybe younger version of bbox) https://github.com/GoogleContainerTools/distroless/issues/183#issuecomment-571723446

**Comment by Cronick** [comment]
Could not find a busybox image with curl. So just aimed for the next best wget.

**Comment by comstud** [comment]
yeah, i saw the same thing last night. This looks like what I'd come up with, but I wanted to give you a chance to fix up your PR. :)  Thanks!

File: docker-compose.yml.example [hunk 1]
```diff
@@ -11,13 +11,12 @@ services:
...
       - "9042:9042"
-# fix after putting curl or wget into image.
-#    healthcheck:
-#      test: curl --fail http://localhost:9042/status || exit 1
-#      interval: 60s
-#      timeout: 30s
-#      retries: 3
-#      start_period: 30s
+    healthcheck:
+      test: ["CMD", "wget", "--no-verbose", "--tries=1", "--spider", "http://localhost:9042/status"]
```
Comment (comstud) [inline docker-compose.yml.example:15]: Hm, somehow we need to ensure this exits with 1 on fail tho (and not another code). Does the method like the commented out curl command not work ?

Comment (Cronick) [reply docker-compose.yml.example:15]: Tried to trigger it with just another port to see, and should behave as expected. At least the best test I could come up with.

{
	"Start": "2024-03-15T15:27:11.521671334Z",
	"End": "2024-03-15T15:27:11.562590376Z",
	"ExitCode": 1,
	"Output": "Connecting to localhost:9043 (127.0.0.1:9043)\nwget: can't connect to remote host (127.0.0.1): Connection refused\n"
}

Comment (comstud) [reply docker-compose.yml.example:15]: Yeah, works for that. And I think this is prob good enough. But just for curiousity, can you see what the exit code is if you plug in a wrong IP address (so it imitates a timeout)?  (Pick a local IP that is not being used by anything)

Comment (Cronick) [reply docker-compose.yml.example:15]: Calling an older version of Fletchling without status endpoint
{
"Start": "2024-03-15T16:18:15.701784382Z",
"End": "2024-03-15T16:18:15.759087799Z",
"ExitCode": 1,
"Output": "Connecting to 192.168.1.15:9042 (192.168.1.15:9042)\nwget: server returned error: HTTP/1.1 404 Not Found\n"

Calling an unused local ip. Still seems to set the container as unhealthy. So not sure how much it matters.
{
"Start": "2024-03-15T16:30:04.221639585Z",
"End": "2024-03-15T16:30:34.261224919Z",
"ExitCode": -1,
"Output": "Health check exceeded timeout (30s): Connecting to 192.168.1.16:9042 (192.168.1.16:9042)\n"
}

Comment (comstud) [reply docker-compose.yml.example:15]: Ok, if it sets the container unhealthy with exit code of -1 there too, then all good. 

Thanks for testing those.

**Review by comstud — APPROVED** [review]

**Pull request merged**